### ISSUE TITLE
bug: ion-checkbox passes incorrect value to components

### ISSUE THREAD

**Issue by queejie** [driftyco/ionic#8738]
**Type**: <span ionic-type>bug</span>

**Ionic Version**: <span ionic-version>2.x</span>

**Platform**: <span ionic-platform>desktop</span>  <span ionic-webview>browser</span>

<span ionic-description>What is displayed is not passed to component methods.

        <ul *ngIf="standard && standard.competencies && standard.competencies.length > 0">
            <li *ngFor="let competency of standard.competencies">
                <ion-checkbox (click)="save(competency.id)" [checked]="isChecked(competency.id)"></ion-checkbox>
                {{competency.id}}::{{competency.body}}
            </li>
        </ul>

In the save(competency.id) call above the last value in the list (4) is passed in each iteration, but the display shows the correct value for each iteration (1,2,3,4).</span>

<span is-issue-template></span>

**Comment by jgw96** [comment]
Hello, thanks for opening an issue with us! Would you mind posting a repo or gist that i can use to reproduce this issue?  Thanks!

**Comment by jgw96** [comment]
Hello all! As it seems it has been a while since there was any activity on this issue i will be closing it for now. Feel free to comment if you are still running into this issue. Thanks for using Ionic!

**Comment by queejie** [comment]
Actually, I replied to by email but that didn't seem to update this ticket.  In any case, I figured out that the problem is the name attribute was not being set uniquely in my loop, which caused the problem.  It is also not related to ionic specifically, but to angular in general.  It was pretty strange symptom to get, so I'm hoping my mistake might help someone else searching for a similar problem.  Thanks.
